src/lib/operators/join_nested_loop_a.hpp

```diff
@@ -12,17 +12,15 @@
 namespace opossum {
 
 /**
- * There are two nested loop joins, implemented by two groups: JoinNestedLoopA and B. They should be functionally
- * identical.
+ * There are two nested loop joins, implemented by two groups: JoinNestedLoopA and B.
+ * Currently only A supports null values.
  *
  * This operator joins two tables using one column of each table.
  * The output is a new table with referenced columns for all columns of the two inputs and filtered pos_lists.
  * If you want to filter by multiple criteria, you can chain this operator.
  *
  * As with most operators, we do not guarantee a stable operation with regards to positions -
  * i.e., your sorting order might be disturbed.
- *
- * Note: JoinNestedLoopA does not support null values in input tables at the moment
  */
 class JoinNestedLoopA : public AbstractJoinOperator {
```

Comment:
From my point of view, there is no reason to have two nested loop joins any longer. The reason was that two groups implemented different joins last semester and the NL join was the first 'warm-up exercise'. So I do not see a good reason to keep both around.

If I do not oversee something, we should take the better/clearer (not the best performance, NL joins suck anyways) and keep only that one around.

Reply:
There are some cases where a NLJ is used in regular databases. I'm fine with dropping one, but it should be the one with better performance and better features (e.g., joining columns with different types).

However, if adding null value support to the other one is also only a question of touching <100 lines, maybe we don't have to make that decision now.

Reply:
I would also like to keep only one NLJ, but after reviewing this, it seems like both implementations are really bloated and based on "old" techniques, like this double visitor pattern. Maybe we should write a new NLJ with iterables?

Reply:
I have also thought about a possible rewrite but as the NestedLoopJoin has a rather low priority I would maybe leave this as is for now. Maybe this could be a project in Develop your own Database as it was last year, just with a more up to date implementation, i.e. with iterables and null value support.

Reply:
I agree, every database needs a NLJ implementation to handle certain cases. I like Timo's idea of removing the old style operators and have a new NLJ.

Reply:
Do we want to have the NUMA join group implement “a modern NLJ” as a warm up in BYOD?
I think it made great sense last semester to do that.